### PR TITLE
Removing extra WITHs in rewriters

### PR DESCRIPTION
**Pull request by sherfert**
This PR

* Removes `ProjectFreshSortExpressions`.
* Changes `normalizeWithClauses` such that it does not introduce new `WITH`s any more.
* Removes `PragmaWithout`.
* Fixes a bug in SemanticAnalysis that relied on the rewriting behavior which was removed.

### CODE REVIEW

File: ast/src/main/scala/org/opencypher/v9_0/ast/semantics/SemanticExpressionCheck.scala [hunk 1]
```diff
@@ -201,11 +200,18 @@ object SemanticExpressionCheck extends SemanticAnalysisTooling {
...
+          case Left(error) =>
+            if (s.declareVariablesToSuppressDuplicateErrors) {
+              // Most of the time we want to suppress if this error occurs again
+              s.declareVariable(x, CTAny.covariant) match {
+                // if the variable is a graph, declaring it will fail
+                case Right(ss) => SemanticCheckResult.error(ss, error)
+                case Left(_error) => SemanticCheckResult.error(s, _error)
+              }
+            } else {
+              // Unless we are ignoring errors anyway in which case this might mess up the scope
```
Comment (Lojjs) [inline ast/src/main/scala/org/opencypher/v9_0/ast/semantics/SemanticExpressionCheck.scala]: I guess the `Unless` in this comment refer back to  the earlier comment `Most of the time...`. But if I read only this line I interpreted it as a warning that the else case might mess up the scope. Maybe something like `If we are ignoring errors anyway, suppressing them might mess up the scope` would be clearer?

File: parser/src/test/scala/org/opencypher/v9_0/ast/semantics/ScopeTreeTest.scala [hunk 2]
```diff
@@ -224,7 +223,7 @@ class ScopeTreeTest extends CypherFunSuite {
     val expected = scope()(
       scope()(),
       scope(intSymbol("p", 10, 36), intSymbol("rng", 25))(),
-      scope(intSymbol("p", 10, 36, 37), typedSymbol("rng", TypeSpec.all, 25, 47))()
+      scope(intSymbol("p", 10, 36, 37))()
```
Comment (Lojjs) [inline parser/src/test/scala/org/opencypher/v9_0/ast/semantics/ScopeTreeTest.scala]: I don't understand why the `typedSymbol`-part disappeared here, can you please explain?

File: rewriting/src/test/scala/org/opencypher/v9_0/rewriting/NormalizeWithClausesTest.scala [hunk 3]
```diff
@@ -576,32 +545,29 @@ class NormalizeWithClausesTest extends CypherFunSuite with RewriteTest with AstC
       """.stripMargin,
     """MATCH (n)
       |WITH n AS n, 0 AS foo
-      |WITH foo AS foo, n AS n
-      |WITH n AS n, foo AS foo, n.bar AS `  FRESHID55` ORDER BY foo, `  FRESHID55`
-      |_PRAGMA WITHOUT `  FRESHID55`
+      |WITH n AS n ORDER BY foo, n.bar
```
[the remark this answers is not in the document]
Comment (sherfert) [reply rewriting/src/test/scala/org/opencypher/v9_0/rewriting/NormalizeWithClausesTest.scala]: Actually, it inserts ` AS n` in the first WITH

File: parser/src/test/scala/org/opencypher/v9_0/ast/semantics/ScopeTreeTest.scala (hunk 2 above)
Comment (sherfert) [reply parser/src/test/scala/org/opencypher/v9_0/ast/semantics/ScopeTreeTest.scala]: The last scope is the one from the RETURN clause. The RETURN does only contain `p`, not `rng`. What I don't understand is why `rng` was in that scope before. It is correct without the `rng`.

File: rewriting/src/test/scala/org/opencypher/v9_0/rewriting/NormalizeWithClausesTest.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (Lojjs) [reply rewriting/src/test/scala/org/opencypher/v9_0/rewriting/NormalizeWithClausesTest.scala]: Ah, my misstake. Thought they looked identical.

**Comment by disolovyov** [comment]
In CfoG we currently rely on `projectFreshSortExpressions`. If this is merged, we would need to rework projection translation (currently there's a regression of 36 tests in `RETURN` and `WITH` acceptance  when running on this branch).

**Comment by disolovyov** [comment]
Ok, so I've managed to fix most of the regression when rebased on this branch, except the whole of `TriadicSelection`. That feature runs queries like this one:

```
MATCH (n:N) 
OPTIONAL MATCH (n)-[r:R]->(m:M) 
WITH n WHERE r IS NULL 
RETURN n
```

And it fails in FE with ```SyntaxException: Variable `r` not defined``` (with default `parsing`/`lateAstRewriting`).

**Comment by sherfert** [comment]
Thanks for the feedback @disolovyov 
The failure you see in `TriadicSelection` is a bug we also see when depending on this branch. I will fix it and then ask you again to run the tests on your side.

**Comment by sherfert** [comment]
@disolovyov I think I should have fixed that bug. If you could rerun your test suite that would be great.

**Comment by disolovyov** [comment]
@sherfert Yep, everything works now, thanks!

**Pull request merged**